optimum/exporters/onnx/input_generators.py

```diff
@@ -0,0 +1,114 @@
+    DTYPE_MAPPER,
+    DummyTextInputGenerator,
+    NormalizedTextConfig,
+)
+
+
+class Gemma3DummyInputGenerator(DummyTextInputGenerator):
+    """Dummy input generator for Gemma3."""
+
+    SUPPORTED_INPUT_NAMES = (
+        "input_ids",
+        "attention_mask",
+        "token_type_ids",
+        "pixel_values",
+    )
+
+    def __init__(
+        self,
+        task: str,
+        normalized_config: NormalizedTextConfig,
+        batch_size: int = DEFAULT_DUMMY_SHAPES["batch_size"],
+        sequence_length: int = DEFAULT_DUMMY_SHAPES["sequence_length"],
+        num_choices: int = DEFAULT_DUMMY_SHAPES["num_choices"],
+        random_batch_size_range: Optional[tuple[int, int]] = None,
+        random_sequence_length_range: Optional[tuple[int, int]] = None,
+        random_num_choices_range: Optional[tuple[int, int]] = None,
+        padding_side: str = "right",
+        preprocessors: list | None = None,
+        **kwargs,
+    ):
+        super().__init__(
+            task,
+            normalized_config,
+            batch_size,
+            sequence_length,
+            num_choices,
+            random_batch_size_range,
+            random_sequence_length_range,
+            random_num_choices_range,
+            padding_side,
+            **kwargs,
+        )
+        self.preprocessors = preprocessors or []
+
+    def generate(
+        self,
+        input_name: str,
+        framework: str = "pt",
+        int_dtype: str = "int64",
+        float_dtype: str = "fp32",
+    ):
+        if self.task in [
+            "image-text-to-text",
+            "image-text-to-text-with-past",
+            "feature-extraction",
+            "feature-extraction-with-past",
+        ]:
+            image = Image.new("RGB", (896, 896), color=128)
+            single_message = [
+                {
+                    "role": "user",
+                    "content": [
+                        {"type": "image", "image": image},
+                        {"type": "text", "text": "Example"},
+                    ],
+                }
+            ]
+        elif self.task in ["text-generation", "text-generation-with-past"]:
+            single_message = [{"role": "user", "content": [{"type": "text", "text": "Example"}]}]
+        else:
+            message = f"The task {self.task} is not supported by the {type(self).__name__}."
+            raise ValueError(message)
+
+        messages = [single_message] * self.batch_size
+        processor = next(
+            (processor for processor in self.preprocessors if isinstance(processor, ProcessorMixin)),
+            None,
+        )
+        if processor is None:
+            message = (
+                "Gemma3 requires an AutoProcessor. Please provide `preprocessors=[AutoProcessor."
+                'from_pretrained(model_name_or_path, padding_side="left")]` to this ONNX config.'
+            )
+            raise ValueError(message)
+
+        inputs = processor.apply_chat_template(
```

Comment:
I would prefer generatoing random dummy inputs, this method doesn't respect the input dimensions in the init method